src/routes/connect/ui/wallets/Wallet.ts

```diff
@@ -0,0 +1,38 @@
+import type IWallet from './interfaces/IWallet';
+import InvalidComponentTypeError from '../../../../lib/errors/InvalidComponentTypeError';
+import Albedo from './Albedo';
+import Freighter from './Freighter';
+import PrivateKey from './PrivateKey';
+import Rabet from './Rabet';
+import XBull from './XBull';
+
+export default class WalletFactory {
+    create(name: string, privateKey?: string): IWallet {
+        let wallet!: IWallet;
+        switch (name) {
+            case 'albedo':
```

Comment:
Each wallet should expose it's name as a constant, so this can then be:

`case Albedo.NAME` and you can avoid the magic string